conclave-host/src/main/kotlin/com/r3/conclave/host/internal/attestation/EnclaveQuoteServiceFactory.kt

```diff
@@ -1,13 +1,19 @@
 package com.r3.conclave.host.internal.attestation
 
 import com.r3.conclave.host.AttestationParameters
+import com.r3.conclave.host.internal.QuotingManager
 
 object EnclaveQuoteServiceFactory {
-    fun getService(attestationParameters: AttestationParameters?): EnclaveQuoteService {
+    fun getService(attestationParameters: AttestationParameters?, quotingManager: QuotingManager): EnclaveQuoteService {
```

Comment:
I would actually get rid of this class and move this method into `EnclaveHost` and then use the instance of the private `enclaveHandle` to determine if gramine or non-gramine DCAP should be used. This will remove the need to have the QuotingManager enum.

Reply:
Interesting. I actually added the enum to AVOID the `is GramineEnclaveHandle`, which seems a bit hacky to me. I will add this condition then